web/src/dropdown_widget.js

```diff
@@ -1,176 +1,271 @@
+        focus_target_on_hidden = true,
+        tippy_props = {},
+        // NOTE: Any value other than `null` will be rendered when class is initialized.
+        default_id = null,
+        unique_id_type = null,
+    }) {
+        this.widget_name = widget_name;
+        this.widget_id = `#${CSS.escape(widget_name)}_widget`;
+        // A widget wrapper may not exist based on the UI requirement.
+        this.widget_wrapper_id = `${this.widget_id}_wrapper`;
+        this.widget_value_selector = `${this.widget_id} .dropdown_widget_value`;
+        this.get_options = get_options;
+        this.item_click_callback = item_click_callback;
+        this.focus_target_on_hidden = focus_target_on_hidden;
+        this.on_show_callback = on_show_callback;
+        this.on_mount_callback = on_mount_callback;
+        this.on_hidden_callback = on_hidden_callback;
+        this.on_exit_with_escape_callback = on_exit_with_escape_callback;
+        this.render_selected_option = render_selected_option;
+        this.tippy_props = tippy_props;
+        this.list_widget = null;
+        this.instance = null;
+        this.default_id = default_id;
+        this.current_value = default_id;
+        this.unique_id_type = unique_id_type;
+        this.$events_container = $events_container;
     }
 
-    tippy.delegate("body", {
-        ...default_popover_props,
-        // Custom theme defined in popovers.css
-        theme: "dropdown-widget",
-        arrow: false,
-        onShow(instance) {
-            instance.setContent(parse_html(render_dropdown_list_container()));
-            const $popper = $(instance.popper);
-            const $dropdown_list_body = $popper.find(".dropdown-list");
-            const $search_input = $popper.find(".dropdown-list-search-input");
-
-            const list_widget = ListWidget.create($dropdown_list_body, get_options(), {
-                name: `${CSS.escape(tippy_props.target)}-list-widget`,
-                get_item: ListWidget.default_get_item,
-                modifier(item) {
-                    return render_dropdown_list({item});
-                },
-                filter: {
-                    $element: $search_input,
-                    predicate(item, value) {
-                        return item.name.toLowerCase().includes(value);
-                    },
-                },
-                $simplebar_container: $popper.find(".dropdown-list-wrapper"),
-            });
-
-            $search_input.on("input.list_widget_filter", () => {
-                const list_items = list_widget.get_current_list();
-                const $no_search_results = $popper.find(".no-dropdown-items");
-                if (list_items.length === 0) {
-                    $no_search_results.show();
-                } else {
-                    $no_search_results.hide();
-                }
-            });
+    init() {
+        if (this.current_value !== null) {
+            this.render();
+        }
```

Comment:
I got this exception when clicking into organization settings just now:

![image](https://github.com/zulip/zulip/assets/2746074/a556694a-28b8-489c-a51b-ce708ee2108d)

Reply:
Fixed.